### DISCUSSION TITLE
Map rollback pegs CPU at 100%

### DISCUSSION THREAD

**Discussion by NotAFile** [piqueserver/piqueserver#652] [Feature Requests]
And takes a loong time... Might be worth looking at the tight loops and putting them into Cython

**Comment by feikname** [comment]
Not sure wheter to tag this as a bug or enhancement.

**Comment by NotAFile** [comment]
Neither really... it's a performance issue

**Comment by 1AmYF** [comment]
>And takes a loong time...

Rollback comes from ancient times when map change was not implemented in aos servers and it was the only way to reset a map or load a new one. IMHO there is no need to invest further time into the script, also it should be removed from the scripts list in the default config.toml as it is really rarely needed at all (only use i can think of right now is the /rollmap command on a build server). Also beside CPU it consumes lots of memory (i guess because non-surface-blocks will be loaded into memory).

**Comment by NotAFile** [comment]
I don't think it's useful to roll back whole maps, or roll over one map with another map, but I believe it can be useful to roll back select parts of the map, for example after excessive destruction or between rounds.
